.github/workflows/_setup_workload_identity.yml

```diff
@@ -0,0 +1,100 @@
+name: Setup Workload Identity 
+
+on:
+  workflow_call:
+    inputs:
+      kubernetes-version: 
+        required: true
+        type: string
+        description: "Kubernetes version to use for AKS cluster creation"
+      cluster-name:
+        required: true
+        description: AKS Cluster Name With Randomly Generated Suffix
+        type: string
+
+  workflow_dispatch:
+    inputs:
+      kubernetes-version:
+        required: true
+        type: string
+        description: "Kubernetes version to use for AKS cluster creation"
+      cluster-name:
+        required: true
+        description: AKS Cluster Name With Randomly Generated Suffix
+        type: string
+
+jobs:
+  setup-workload-identity:
+    name: Setup Workload Identity 
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Log into Azure
+        run: |
+          az login --service-principal \
+            --username ${{ secrets.AZURE_SERVICE_PRINCIPAL_APP_ID }} \
+            --password ${{ secrets.AZURE_SERVICE_PRINCIPAL_PASSWORD }} \
+            --tenant ${{ secrets.AZURE_SERVICE_PRINCIPAL_TENANT }}
+            
+      - name: Install Dependencies 
+        id: install-dependencies 
+        run: |
+            # curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+            # chmod +x kubectl
+            # sudo mv kubectl /usr/local/bin
+            # kubectl version 2>&1 || true
+
+      - name: Setup Workload Identity and Kafka Cluster
+        env:
+          SERVICE_ACCOUNT_NAMESPACE: ${{ vars.SERVICE_ACCOUNT_NAMESPACE }}
```

Comment:
These could be auto-generated instead of configured.

Reply:
Service account namespace has to be in kafka namespace. I see the point of removing federated credential name and service account name from the repo scoped variable list because of simplicity of maintenance. We might as well just bake these three things into the code. Their names don't matter, and they are gone with cluster cleanup.